Examples/Python/tests/test_examples.py

```diff
@@ -135,6 +135,54 @@ def test_fatras(trk_geo, tmp_path, field, assert_root_hash):
         assert_root_hash(f, rfp)
 
 
+@pytest.mark.slow
+@pytest.mark.skipif(not geant4Enabled, reason="Geant4 not set up")
+@pytest.mark.skipif(not dd4hepEnabled, reason="DD4hep not set up")
+def test_geant4(tmp_path, field, assert_root_hash):
+    # This test literally only ensures that the full chain example can run without erroring out
+    getOpenDataDetector(
+        getOpenDataDetectorDirectory()
+    )  # just to make sure it can build
+
+    csv = tmp_path / "csv"
+    csv.mkdir()
+
+    root_files = [
+        "fatras_particles_final.root",
+        "fatras_particles_initial.root",
+        "hits.root",
+    ]
+
+    assert len(list(csv.iterdir())) == 0
+    for rf in root_files:
+        assert not (tmp_path / rf).exists()
+
+    script = (
+        Path(__file__).parent.parent.parent.parent
+        / "Examples"
+        / "Scripts"
+        / "Python"
+        / "geant4.py"
+    )
+    assert script.exists()
+    env = os.environ.copy()
+    env["ACTS_LOG_FAILURE_THRESHOLD"] = "WARNING"
+    subprocess.check_call(
+        [str(script)],
+        cwd=tmp_path,
+        env=env,
+        stderr=subprocess.STDOUT,
+    )
+
+    assert_csv_output(csv, "particles_final")
+    assert_csv_output(csv, "particles_initial")
+    assert_csv_output(csv, "hits")
+    for f in root_files:
+        rfp = tmp_path / f
+        assert rfp.exists()
+        assert rfp.stat().st_size > 2**10 * 10
```

Comment:
Could we add has checks on these? I *think* the G4 output should be deterministic here.

Reply:
I tried to use the fatras checks but it failed at the `hits`. the CI produced different hits than I had locally :disappointed:

Reply:
That's ok though. We have this in other cases too, we just need to pull the reference from the CI in that case.

Reply:
you mean using `assert_root_hash` right?
in that case the unit tests will fail locally right? happy to add it if that is expected

Reply:
The hash checks are off by default, and there's a mechanism to supply a local hash file with references. See the [docs page](https://acts.readthedocs.io/en/latest/examples/python_bindings.html#running-the-hash-checks-locally-and-how-to-update-the-reference-hashes)

Reply:
ah nice will do. the other checks from fatras I had to remove because it produced different results locally and in the CI